optimism/src/keccak/helpers.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{
+    keccak::interpreter::Interpreter,
+    lookups::{Lookup, LookupTableIDs::*},
+};
+use ark_ff::{One, Zero};
+use std::fmt::Debug;
+
+/// This trait contains helper functions for the lookups used in the Keccak circuit
+/// using the zkVM lookup tables
+pub trait LogupHelpers<F: One + Debug + Zero>
+where
+    Self: Interpreter<F>,
+{
+    /// Adds a lookup to the RangeCheck16 table
+    fn lookup_rc16(&mut self, flag: Self::Variable, value: Self::Variable) {
+        self.add_lookup(flag, Lookup::read_one(RangeCheck16Lookup, vec![value]));
+    }
+
+    /// Adds a lookup to the Reset table
+    fn lookup_reset(
+        &mut self,
+        flag: Self::Variable,
+        dense: Self::Variable,
+        sparse: Self::Variable,
+    ) {
+        self.add_lookup(flag, Lookup::read_one(ResetLookup, vec![dense, sparse]));
+    }
+
+    /// Adds a lookup to the Shift table
+    fn lookup_sparse(&mut self, flag: Self::Variable, value: Self::Variable) {
+        self.add_lookup(flag, Lookup::read_one(SparseLookup, vec![value]));
+    }
+
+    /// Adds a lookup to the Byte table
+    fn lookup_byte(&mut self, flag: Self::Variable, value: Self::Variable) {
+        self.add_lookup(flag, Lookup::read_one(ByteLookup, vec![value]));
+    }
+
+    /// Adds a lookup to the Pad table
+    fn lookup_pad(&mut self, flag: Self::Variable, value: Vec<Self::Variable>) {
+        self.add_lookup(flag, Lookup::read_one(PadLookup, value));
+    }
+
+    /// Adds a lookup to the RoundConstants table
+    fn lookup_round_constants(&mut self, flag: Self::Variable, value: Vec<Self::Variable>) {
+        self.add_lookup(flag, Lookup::read_one(RoundConstantsLookup, value));
+    }
+
+    fn read_syscall(&mut self, flag: Self::Variable, value: Vec<Self::Variable>) {
+        self.add_lookup(flag, Lookup::read_one(SyscallLookup, value));
+    }
+
+    fn write_syscall(&mut self, flag: Self::Variable, value: Vec<Self::Variable>) {
+        self.add_lookup(flag, Lookup::write_one(SyscallLookup, value));
+    }
+}
+
+/// This trait contains helper functions for boolean operations used in the Keccak circuit
+pub trait BoolHelpers<F: One + Debug + Zero>
```

Comment:
I don't know what's the good practice, but in general I find it very counterintuitive to define actual functions within traits. If `BoolHelpers` is not going to be instantiated especially... I kind of think every trait should have several instantiations, but `BoolHelpers` does not need any. Weird.

However I don't see a good alternative for now. One could declare these functions (`is_boolean` and so on) on the top level, but it seems to be inelegant in a different way. So I probably agree to go forward with this technique.